app/src/main/java/com/willowtree/vocable/presets/adapter/CustomCategoryAdapter.kt

```diff
@@ -0,0 +1,72 @@
+package com.willowtree.vocable.presets.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isInvisible
+import androidx.recyclerview.widget.RecyclerView
+import com.willowtree.vocable.R
+import com.willowtree.vocable.databinding.CustomCategorySwitchItemBinding
+import com.willowtree.vocable.room.Category
+import com.willowtree.vocable.utils.LocalizedResourceUtility
+import org.koin.core.KoinComponent
+import org.koin.core.inject
+
+class CustomCategoryAdapter(
+    private var categories: Map<Category, Boolean>,
+    private val numRows: Int,
+    private val onCategoryToggle: (Category, Boolean) -> Unit
+) : RecyclerView.Adapter<CustomCategoryAdapter.CustomCategoryViewHolder>(), KoinComponent {
+
+    private val localizedResourceUtility: LocalizedResourceUtility by inject()
+
+    private var _minHeight: Int? = null
+
+    inner class CustomCategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        private val binding: CustomCategorySwitchItemBinding =
+            CustomCategorySwitchItemBinding.bind(itemView)
+
+        fun bind(category: Category, isChecked: Boolean, onCategoryToggle: (Category, Boolean) -> Unit) {
+            binding.categoryText.text = localizedResourceUtility.getTextFromCategory(category)
+
+            binding.categoryContainer.action = { binding.toggleSwitch.isChecked = !binding.toggleSwitch.isChecked }
+
+            binding.toggleSwitch.isChecked = isChecked
+
+            binding.toggleSwitch.setOnCheckedChangeListener { buttonView, isChecked ->
+                onCategoryToggle(category, isChecked)
+            }
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomCategoryViewHolder {
+        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.custom_category_switch_item, parent, false)
+        itemView.isInvisible = true
+        parent.post {
+            with(itemView) {
+                findViewById<View>(R.id.filler_view).minimumHeight = getMinHeight(parent)
+                isInvisible = false
+            }
+        }
+
+        return CustomCategoryViewHolder(itemView)
+    }
```

Comment:
Wouldn't it be easier to just pass in `CustomCategorySwitchItemBinding` into the `CustomCategoryViewHolder ` class? You could handle all your initial view stuff in there too since you have access to the `binding` val.

Reply:
So you could do something like
```
override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding = CustomCategorySwitchItemBinding.inflate(inflater, parent, false)

       // do your init binding stuff here

        return ViewHolder(binding)
    }
....

// Then in your view holder you'd just have reference to your `binding` already to use for the `bind()` method.
```